openapi.yaml

```diff
@@ -471,6 +591,173 @@ paths:
+      properties:
+        base:
+          $ref: '#/components/schemas/Base'
+        channel:
+          type: string
+        charm:
+          type: string
+        constraints:
+          $ref: '#/components/schemas/Constraints'
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        exposed:
+          type: boolean
+        exposed-endpoints:
+          type: object
+          patternProperties:
+            .*:
+              $ref: '#/components/schemas/ExposedEndpoint'
+        life:
+          type: string
+        principal:
+          type: boolean
+        remote:
+          type: boolean
+        tag:
+          type: string
+      additionalProperties: false
+      required:
+        - tag
+        - principal
+        - exposed
+        - remote
+        - life
+    ApplicationSetCharm:
+      type: object
+      properties:
+        application:
+          type: string
+        channel:
+          type: string
+        charm-origin:
+          $ref: '#/components/schemas/CharmOrigin'
+        charm-url:
+          type: string
+        config-settings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        config-settings-yaml:
+          type: string
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        force:
+          type: boolean
+        force-base:
+          type: boolean
+        force-units:
+          type: boolean
+        generation:
+          type: string
+        resource-ids:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        storage-constraints:
+          type: object
+          patternProperties:
+            .*:
+              $ref: '#/components/schemas/StorageConstraints'
+      additionalProperties: false
+      required:
+        - application
+        - generation
+        - charm-url
+        - channel
+        - force
+        - force-units
+        - force-base
+    Base:
+      type: object
+      properties:
+        channel:
+          type: string
+        name:
+          type: string
+      additionalProperties: false
+      required:
+        - name
+        - channel
+    CharmRelation:
+      type: object
+      properties:
+        interface:
+          type: string
+        limit:
+          type: integer
+        name:
+          type: string
+        optional:
+          type: boolean
+        role:
+          type: string
+        scope:
+          type: string
+      additionalProperties: false
+      required:
+        - name
+        - role
+        - interface
+        - optional
+        - limit
+        - scope
+    CharmOrigin:
+      type: object
+      properties:
+        architecture:
+          type: string
+        base:
+          $ref: '#/components/schemas/Base'
```

Comment:
I think if we remove this as I suggested above, then we might just need a `channel` field in here that is a `CharmChannel` object containing track/channel/branch?

Reply:
So it's important to make a terminology distinction here. 

In Juju, we have a [series.Channel](https://github.com/juju/juju/blob/2b2308500d71d17867256cc81dce9dbce877bcc4/core/series/channel.go#L40) that the [Base](https://github.com/juju/juju/blob/2b2308500d71d17867256cc81dce9dbce877bcc4/core/series/base.go#L15) uses, which looks something like `22.04@stable` (track/risk -- doesn't have branch). And we also have a [charm.Channel](https://github.com/juju/charm/blob/6beed86562dc206e5e53aeb168eb94aeaf4d6729/channel.go#L40) that looks like  `latest/stable/br`, which was the prominent "channel" in the past. Lately we have been moving away from using the charm.Channel and using mostly the Base (which also has the OS info). So I suggest sticking with that. At the very least, I think we should refrain from using two different `channel`s, and pick just one of them. Otherwise we'll inherit this nasty channel overload.

Reply:
Well it's not necessarily that easy I don't think - for a given application deployment you have **both** a base channel, and a charm channel at play?

Reply:
Oh you're right, disregard my comment above, the channel information (for both types of channels) has to be there. Though I will suggest not to call both of them `channel`s.